### PR TITLE
HTTP event listener cleanups

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Comment by losipiuk** [comment]
cc: @mosiac1

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 1]
```diff
@@ -162,6 +162,18 @@ public void onFailure(Throwable t)
                 (long) delay.getValue(), delay.getUnit());
     }
 
+    private boolean shouldRetry(StatusResponse result)
+    {
+        int statusCode = result.getStatusCode();
+        if (statusCode >= 200 && statusCode < 300) {
+            return false; // OK
+        }
+        if (statusCode >= 400 && statusCode < 500) {
```
Comment (losipiuk) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:171]: actually I am not convinced that we should retry anything other than 5xx, as it was originally. Is there any reason for other codes @mosiac1 .

Thanks @hashhar for pointing this out :)

Comment (mosiac1) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:171]: my reasoning was to help prevent dropped events - if there is no retry then that event is lost. in some systems codes other than 5xx can be resolved and are due to a middleman issue (like kubernetes or a gateway).

Comment (hashhar) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:171]: 408 Request Timeout and 429 Too Many Requests are the ones that I know people retry. Other 4xx seem like they'd keep failing e.g. 403, 404, 400 etc.

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 2]
```diff
@@ -162,6 +162,18 @@ public void onFailure(Throwable t)
...
+            return false; // 4xx is not retriable
```
Comment (hashhar) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:172]: would a debug log be useful here?

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 3]
```diff
@@ -124,7 +124,7 @@ public void onSuccess(StatusResponse result)
                             {
                                 verify(result != null);
 
-                                if (!(result.getStatusCode() >= 200 && result.getStatusCode() < 300) && attempt < config.getRetryCount()) {
+                                if (shouldRetry(result) && attempt < config.getRetryCount()) {
```
Comment (mosiac1) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:127]: i think we should split this into 2 ifs, so when `shouldRetry` is true but there are no retries left it is communicated

Comment (mosiac1) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:127]: this is something i should have implemented, sorry about that

**Comment by mosiac1** [comment]
Hello, I'm working on a fix for https://github.com/airlift/airlift/issues/983 which is affecting the event listener.

If it's ok with everyone I can include these changes in that PR so there won't be conflicts. If not that is also fine with me.

**Pull request closed without merging**